test/functional/feature_staking_unlock_and_withdraw.py

```diff
@@ -87,9 +88,16 @@ def run_test(self):
         assert_equal(events[2].value['event_id'], 'StakeUnlocked')
 
         ledger = list(client.get_staking_ledger())
+
         assert_equal(len(ledger),2)
 
-        time.sleep(500)
+        num_attempts = 0
+        max_attempts = 10000
+        while client.current_era() < client.withdrawal_era(alice):
+            num_attempts += 1
```

Comment:
To make this in terms of time, I'd use a period instead of num_attempts, and then wait that period in the sleep. When max_wait is exceeded, I'd call it an error. Since Python is very tolerant with types, you can make use of it and have only time instead of time + unitless numeric attempts.

Reply:
@TheQuantumPhysicist Is "error" different from "Exception"?